src/libcore/slice/mod.rs

```diff
@@ -400,14 +400,15 @@ impl<T> SliceExt for [T] {
         while size > 1 {
             let half = size / 2;
             let mid = base + half;
-            // mid is always in [0, size].
+            // mid is always in [0, size).
+            // (that means mid is >= 0 and < size)
```

Comment:
Very minor style nit, but since the lines below are effectively subordinate bullet points, I think it would look nicer if this was one line. Fix if you want, or don't.